flask_of_oil/oauth_filter.py

```diff
@@ -111,16 +115,39 @@ def _extract_access_token(incoming_request=None):
 
         return query_param_access_token
 
-    def _authorize(self, token_claims, endpoint_scopes, endpoint_claims):
+    def _authorize(self, token_claims, endpoint_scopes, endpoint_claims, endpoint_audience):
+
+        if endpoint_audience is not None:
+            if not isinstance(endpoint_audience, list):
+                endpoint_audience = [endpoint_audience]
+
+            if not isinstance(token_claims['aud'], list):
+                token_claims['aud'] = [token_claims['aud']]
+
+            if len(set(endpoint_audience)&set(token_claims['aud'])) == 0:
+                return False
 
         if endpoint_claims is None:
             endpoint_claims = {}
 
         for claim in endpoint_claims:
-            if claim not in token_claims or \
-                    (endpoint_claims[claim] is not None and token_claims[claim] != endpoint_claims[claim]):
+            if claim not in token_claims:
                 return False
 
+            if endpoint_claims[claim] is not None:
```

Comment:
Does it make sense to factor this out? It's getting to be a lot of nesting here.